pkg/apis/pipeline/v1/task_validation.go

```diff
@@ -406,6 +407,17 @@ func validateTaskContextVariables(ctx context.Context, steps []Step) *apis.Field
 	return errs.Also(validateVariables(ctx, steps, "context\\.task", taskContextNames))
 }
 
+func validateTaskResultsVariables(ctx context.Context, steps []Step, results []TaskResult) (errs *apis.FieldError) {
```

Comment:
Can you add a docstring for this function?

Reply:
sure! Thanks for the suggestion!
I found it hard to understand some functions in this file cuz they lack of docstrings